includes/template-library/manager.php

```diff
@@ -200,15 +204,18 @@ public function get_source( $id ) {
 	 *
 	 * Retrieve all the templates from all the registered sources.
 	 *
-	 * @since 1.0.0
-	 * @access public
+	 * @param array $only_sources
 	 *
-	 * @return array Templates array.
+	 * @return array
 	 */
-	public function get_templates() {
+	public function get_templates( $only_sources = [] ) {
 		$templates = [];
 
 		foreach ( $this->get_registered_sources() as $source ) {
+			if ( ! empty( $only_sources ) && ! in_array( $source->get_id(), $only_sources, true ) ) {
```

Comment:
Just a suggestion: use the Collection `filter` method to remove the sources if needed and then use the method `flatten ` which does not exist yet in our code but can be added to the collection: 
https://laravel.com/docs/8.x/collections#method-flatten
https://github.com/illuminate/collections/blob/a9a225f0378fe64204b09e5897c5950605bc8dd4/Arr.php#L211

Reply:
Would be great, but I think guess it will be out of scope. Maybe we need to create a task for that.